packages/components/resource-pages/list-page.tsx

```diff
@@ -0,0 +1,263 @@
+import { getName } from "../../utils/selectors/k8s";
+import "./list-page.scss";
+import { bucketClassResource, nameSpaceStoreResource } from "../resources";
+
+const tableColumnInfo = [
+  { className: "", id: "name" },
+  {
+    className: "",
+    id: "status"
+  },
+  {
+    className: "",
+    id: "provider"
+  },
+  {
+    className: "",
+    id: "buckets"
+  },
+  { className: "dropdown-kebab-pf pf-c-table__action", id: "" }
+];
+
+type ResourceTableProps = {
+  data: K8sResourceCommon[];
+  unfilteredData: K8sResourceCommon[];
+  loaded: boolean;
+  loadError: any;
+  rowData: CustomData;
+};
+
+const ResourceTable: React.FC<ResourceTableProps> = (props) => {
+  const { t } = useTranslation();
+  const tableColumns = React.useMemo<TableColumn<K8sResourceCommon>[]>(
+    () => [
+      {
+        title: t("Name"),
+        sort: "metadata.name",
+        transforms: [sortable],
+        props: {
+          className: tableColumnInfo[0].className
+        },
+        id: tableColumnInfo[0].id
+      },
+      {
+        title: t("Status"),
+        props: {
+          className: tableColumnInfo[1].className
+        },
+        id: tableColumnInfo[1].id
+      },
+      {
+        title: t("Provider"),
+        props: {
+          className: tableColumnInfo[2].className
+        },
+        id: tableColumnInfo[2].id
+      },
+      {
+        title: t("Buckets"),
+        props: {
+          className: tableColumnInfo[3].className
+        },
+        id: tableColumnInfo[3].id
+      },
+      {
+        title: "",
+        props: {
+          className: tableColumnInfo[4].className
+        },
+        id: tableColumnInfo[4].id
+      }
+    ],
+    [t]
+  );
+
+  const [columns] = useActiveColumns({
+    columns: tableColumns,
+    showNamespaceOverride: false,
+    columnManagementID: null
+  });
+  return (
+    <VirtualizedTable
+      {...props}
+      aria-label={t("Resource Page")}
+      columns={columns}
+      Row={RowRenderer}
+    />
+  );
+};
+
+type CustomData = {
+  launchModal: LaunchModal;
+  kebabActions?: (t: TFunction) => { [key: string]: { value: string } };
+  bucketPolicyMap: { [key: string]: { value: string } };
+};
+
+const RowRenderer: React.FC<RowProps<NamespaceStoreKind, CustomData>> = ({
+  obj,
+  activeColumnIDs,
+  rowData
+}) => {
+  const { t } = useTranslation();
+  const { launchModal, kebabActions, bucketPolicyMap } = rowData;
+  return (
+    <>
+      <TableData {...tableColumnInfo[0]} activeColumnIDs={activeColumnIDs}>
+        {getName(obj)}
+      </TableData>
+      <TableData {...tableColumnInfo[1]} activeColumnIDs={activeColumnIDs}>
+        <OperandStatus operand={obj} />
+      </TableData>
+      <TableData {...tableColumnInfo[2]} activeColumnIDs={activeColumnIDs}>
+        {NS_PROVIDER_REVERSE_MAP[obj.spec.type]}
+      </TableData>
+      <TableData {...tableColumnInfo[3]} activeColumnIDs={activeColumnIDs}>
+        {String(bucketPolicyMap[obj.metadata.name] || 0) + " " + t("Buckets")}
```

Comment:
store count in a const variable on top (`const bucketsCount = bucketPolicyMap[obj?.metadata?.name] || 0`), not need to change to `String`.
```suggestion
          {t('{{bucketsCount}} Buckets', {
            bucketsCount: bucketsCount,
          })}
```